test/common/json/gen_excluded_unicodes.cc

```diff
@@ -0,0 +1,127 @@
+#include "source/common/protobuf/utility.h"
+
+#include "test/common/json/utf8.h"
+
+#include "absl/strings/str_format.h"
+
+namespace Envoy {
+namespace Json {
+namespace {
+
+// Collects unicode values that cannot be handled by the protobuf json encoder.
+// This is not needed for correct operation of the json sanitizer, but it is
+// needed for comparing sanitization results against the proto serializer, and
+// for differential fuzzing. We need to avoid comparing sanitization results for
+// strings containing utf-8 sequences that protobufs cannot serialize.
+//
+// bazel -c opt run test/common/json:gen_excluded_unicodes |& grep -v 'contains invalid UTF-8'
+//
+// The grep pipe is essential as otherwise you will be buried in thousands of
+// messages from the protobuf library that cannot otherwise be trapped. The
+// "-c opt" is recommended because otherwise it takes almost a minute to run
+// through all 4-byte UTF-8 sequences.
+//
+// Running in this mode causes two tests to fail, but prints two initialization
+// blocks for invalid byte code ranges, which can then be pasted into the
+// InvalidUnicodeSet constructor in json_sanitizer_test_util.cc.
+//
+// The output of this command can then be pasted into the InvalidUnicodeSet
+// constructor for json_sanitizer_test_util.cc. It should only be necessary
+// to re-compute the invalid set if the protobuf library changes.
```

Comment:
Please add pointers to the repository_locations.bzl for visibility to protobuf upgraders.